crypto/fipsmodule/aes/aes_nohw.c

```diff
@@ -338,15 +338,15 @@ static inline void aes_nohw_compact_block(aes_word_t out[AES_NOHW_BLOCK_WORDS],
 #if defined(OPENSSL_SSE2)
   // No conversions needed.
 #elif defined(OPENSSL_64_BIT)
-  uint64_t a0 = aes_nohw_compact_word(out[0]);
```

Comment:
I noticed that you have added `CRYPTO_bswap8_le` to every invocation of `aes_nohw_compact_word`. Instead, you could move the `CRYPTO_bswap8_le` into `aes_nohw_compact_word` itself.

Further, if you look at `aes_nohw_compact_word`, you can see that `aes_nohw_compact_word` looks like this:

```
  // Numbering the 64/2 = 16 4-bit chunks, least to most significant, we swap
  // quartets of those chunks:
  //   0 1 2 3 | 4 5 6 7 | 8  9 10 11 | 12 13 14 15 =>
  //   0 2 1 3 | 4 6 5 7 | 8 10  9 11 | 12 14 13 15
  a = aes_nohw_delta_swap(a, UINT64_C(0x00f000f000f000f0), 4);
  // Swap quartets of 8-bit chunks (still numbering by 4-bit chunks):
  //   0 2 1 3 | 4 6 5 7 | 8 10  9 11 | 12 14 13 15 =>
  //   0 2 4 6 | 1 3 5 7 | 8 10 12 14 |  9 11 13 15
  a = aes_nohw_delta_swap(a, UINT64_C(0x0000ff000000ff00), 8);
  // Swap quartets of 16-bit chunks (still numbering by 4-bit chunks):
  //   0 2 4 6 | 1  3  5  7 | 8 10 12 14 | 9 11 13 15 =>
  //   0 2 4 6 | 8 10 12 14 | 1  3  5  7 | 9 11 13 15
  a = aes_nohw_delta_swap(a, UINT64_C(0x00000000ffff0000), 16);
```
So if you really wanted to, I think you might be able to change the inputs to the aes_nohw_delta_swap calls to optimize out the `CRYPTO_bswap8_le` call completely. If you want to save this change for the future, we can defer this change.

But at a minimum, let's move the byte swapping to `aes_nohw_compact_word` to make the diff from BoringSSL smaller.

Reply:
Given your later comments, I've now changed this to use CRYPTO_load/store instead.  I think this makes the rest of the comments here obsolete - please let me know if you'd like me to do anything different here.

Reply:
Hi, I would rather we keep this code exactly as it is in BoringSSL as much as possible and instead add the `CRYPTO_bswap` call to the top of `aes_nohw_compact_word` along with a`"// TODO: merge byte swapping into delta swapping?` comment, and the same for `uncompact_word`.  This way we ensure we're not impeding the performance at all on the already-supported targets. I guess this is the only two places where we'll need to use `CRYPTO_bswap` outside internal.h, and hopefully it is temporary (NBD if not).